homeassistant/components/derivative/sensor.py

```diff
@@ -114,6 +115,25 @@ def calc_derivative(entity, old_state, new_state):
             ):
                 return
 
+            now = new_state.last_updated
+            self._state_list.append((now, new_state.state))
+
+            # Get indices of tuples that are older than `unit_time`
+            to_remove = []
+            for i, (timestamp, _) in enumerate(self._state_list):
```

Comment:
I can replace this logic with
```
            self._state_list = [
                (timestamp, state)
                for (timestamp, state) in self._state_list
                if (now - timestamp).total_seconds() < self._unit_time
            ]
```
but it would be less efficient, but that might not matter?